### ISSUE TITLE
Bug: Recipe consumes wrong amount of child products

### ISSUE THREAD

**Issue by zaped212** [grocy/grocy#1118]
I thought I saw a previous bug over this topic, but when I tried to re-find it I could not. So I apologize if this if a duplicate of another open/closed ticket.

I have been working on setting up a server that has "Generic Products" as the products that will be used in the recipes ( eg: Soy Sauce ).

This for example is in fluid ounces.
I setup a child product that has stock in bottles.
There is a conversion from bottles to fluid ounces. For argument sake lets assume 100 fluid ounces per bottle.

In stock I have 10 bottles ( 1,000 fluid ounces ) of the multi pack ( child ) product.

I have a recipe setup to consume 10 fluid ounces of soy sauce ( parent product ).

When I consume the recipe it ends up consuming 10 bottles on the child product instead of only 10 ounces.

I create https://demo.grocy.info/recipes?recipe=15 to show this.

**Comment by berrnd** [comment]
Thanks for reporting, have not looked into it yet - here is a snapshot of the current demo database as it will get dropped/recreated every night.

[grocy_en.zip](https://github.com/grocy/grocy/files/5484104/grocy_en.zip)

**Comment by berrnd** [comment]
This is now resolved, applying unit conversions when consuming substitution products was just not handled at all so far.

Product specific unit conversions for child/parent product relations need to be defined at the child product - which you did, but the conversion `Fluid Ounce -> Bottle` was missing (just the other way round was there).

In the next version there will also be an option to create those inverse conversion automatically when adding a conversion:

![grafik](https://user-images.githubusercontent.com/2161815/101395572-6db41600-38ca-11eb-8ec0-08615bda17dd.png)

I took the database snapshot from above and imported it into a private demo instance ([ref](https://www.reddit.com/r/grocy/comments/jwnk9b)) of the pre-release demo, if you want to play around with the existing example:
=> https://test-vtqalyegly1it542dmykk.demo-prerelease.grocy.info/recipes?recipe=15

Also related to #899.
